### ISSUE TITLE
The exit  command does not work if the user's default shell is zsh

### ISSUE THREAD

**Issue by MOIS3Y** [nix-community/nix-on-droid#394]
The `exit` command does not work if the user's default shell is zsh
The process just hangs, to close the tab or stop the application you need to call the context menu and click kill the process

Is there a way to figure out what the problem is?
I'm using the home manager [module](https://github.com/MOIS3Y/nixos-configurations/blob/main/homes/_shared/programs/zsh.nix) (but not sure if this could be the problem)

```nix
{ pkgs, lib, ... }: {
  user.shell = "${lib.getExe pkgs.zsh}";
}
```

**Comment by t184256** [comment]
does it have leftover background processes? I remember having something like that

**Comment by MOIS3Y** [comment]
> does it have leftover background processes? I remember having something like that

Thanks for the quick response!
I don’t think so, this is what the list of processes looks like after the first launch of Nix-on-droid

<details>
  <summary>Screenshot</summary>
  <img src="https://github.com/user-attachments/assets/03b5024f-017f-4c1d-8009-689fd85c93fa">
</details>

There are lf and ssh-agent processes there, but the problem was relevant earlier when there were no packages other than the zsh module

**Comment by t184256** [comment]
Can't reproduce with just the default flake bootstrap + `user.shell = "${pkgs.zsh}/bin/zsh";`, so my suspicions fall on ssh-agent.

**Comment by Gerschtli** [comment]
Looks like there is a ssh-agent running that might need manual killing IIRC.

**Comment by MOIS3Y** [comment]
@t184256 yes you are right, after rebuild new generation without ssh-agent (omz plugin) the exit cmd works.
I think that the issue can be closed, I will look for a way to start the agent manually, thanks for the help

**Comment by amalgame21** [comment]
It happens to me when it is still connected from PC to nix-on-droid via ssh.
I have to both CTRL-C the sshd in nix-on-droid and disconnect from PC to make it exit properly, just CTRL-C sshd won't make PC disconnect.
